### PR TITLE
Remove unwrap in deframe_one

### PR DESCRIPTION
**Pull request by ab-cgh**

### CODE REVIEW

File: rustls/src/msgs/hsjoiner.rs [hunk 1]
```diff
@@ -90,16 +90,15 @@ impl HandshakeJoiner {
...
             let mut rd = codec::Reader::init(&self.buf);
-            let payload = HandshakeMessagePayload::read_version(&mut rd, version);
-
-            if payload.is_none() {
-                return false;
-            }
-
+            let payload = match HandshakeMessagePayload::read_version(&mut rd, version) {
+                Some(p) => p,
+                None => return false,
```
Comment (briansmith) [inline rustls/src/msgs/hsjoiner.rs:95]: Normally I do:
```
=> {
    return false;
}
```

Although that's partly because rustfmt forces me to, I think.

Comment (ab-cgh) [reply rustls/src/msgs/hsjoiner.rs:95]: Stable rustfmt didn't change this for me. I don't think rustfmt 2 will either, but I'm unable to install it currently to test that.

**Review by briansmith — APPROVED** [review]

File: rustls/src/msgs/hsjoiner.rs (hunk 1 above)
Comment (briansmith) [reply rustls/src/msgs/hsjoiner.rs:95]: I see some other instances in the codebase where this formatting is used so no need to change anything.

**Review by djc — APPROVED** [review]
Looks good, thanks!

**Pull request merged**